### ISSUE TITLE
[Admin][Product] Unable to add a simple product

### ISSUE THREAD

**Issue by psihius** [Sylius/Sylius#8102]
At current master, unable to add a product via "New" -> "Simple" due to the fact that `position` field is `null` on SQL query, that results in "Cannot be null" error.

>  request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\NotNullConstraintViolationException: "An exception occurred while executing 'INSERT INTO sylius_product_variant (code, created_at, updated_at, position, on_hold, on_hand, tracked, width, height, depth, weight, shipping_required, product_id, tax_category_id, shipping_category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["testproduct", "2017-05-25 11:14:28", "2017-05-25 11:14:28", null, 0, 0, 0, null, null, null, null, 1, 128, 1, null]:  SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null" at /var/www/store.dzintars.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 112 {"exception":"[object] (Doctrine\\DBAL\\Exception\\NotNullConstraintViolationException(code: 0): An exception occurred while executing 'INSERT INTO sylius_product_variant (code, created_at, updated_at, position, on_hold, on_hand, tracked, width, height, depth, weight, shipping_required, product_id, tax_category_id, shipping_category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [\"testproduct\", \"2017-05-25 11:14:28\", \"2017-05-25 11:14:28\", null, 0, 0, 0, null, null, null, null, 1, 128, 1, null]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null at /var/www/store.dzintars.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:112, Doctrine\\DBAL\\Driver\\PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null at /var/www/store.dzintars.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:93, PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null at /var/www/store.dzintars.com/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:91)"} []

This should have some default value probably.

**Comment by kochen** [comment]
@psihius does this help?
```ALTER TABLE sylius_product_variant ALTER COLUMN position SET DEFAULT 0;```

**Comment by psihius** [comment]
@kochen yep, this of course helps, but if we fix it this way, the mapping has to be changed in the project, because Doctrine does not allow to override field definitions at all :( 
And the question is - does the Sortable extension allow nulls at all? It may break, so maybe the proper decision would be to do a `$productVariant->setPosition(1)` when creating a product variant

**Comment by lchrusciel** [comment]
@psihius this logic [is tested](https://github.com/Sylius/Sylius/blob/master/features/product/managing_products/adding_product.feature#L13-L21) and the tests are green. Can you provide some more details about your problem or try to discover between your problem and behat scenario?

**Comment by psihius** [comment]
@lchrusciel Okay, there is a small quirk with it. If you just enter the `code`, `name` and `slug` - everything works.

But when you also check a channel and set the `price` for that channel - then it gives you an error.

**Comment by lchrusciel** [comment]
@psihius Can you provide a behat scenario which fails? I would be much easier to tackle a problem.

**Comment by Zales0123** [comment]
Is it still an issue? I've tried to reproduce it on the latest master and it works fine for me. /cc @psihius @kochen @lchrusciel

**Comment by pjedrzejewski** [comment]
I tried this as well and works. Please reopen if still a bug for you. :)

**Comment by nasserman** [comment]
I have got same error, I installed ver. 1.9.1. just after installing, go to add new simple product, but i got this error:

`An exception occurred while executing 'INSERT INTO sylius_product_variant (code, created_at, updated_at, position, enabled, product_id, on_hold, on_hand, tracked, width, height, depth, weight, shipping_required, tax_category_id, shipping_category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["pen", "2021-03-21 11:14:07", null, null, 1, 5, 0, 0, 0, null, null, null, null, 1, null, null]:`

`SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null`

In the DB, position default value is set on 0.

**Comment by camilleislasse** [comment]
Same here

Projet installed without sample data

![2021-03-22_11-58](https://user-images.githubusercontent.com/34024380/111979992-02532480-8b06-11eb-8b1b-f3c3d471b776.png)

**Comment by nasserman** [comment]
@pjedrzejewski can you check this please?

**Comment by pamil** [comment]
See https://github.com/Sylius/Sylius/pull/12468, adding this conflict helps and new Sylius patch releases are coming in an hour or so 🎉

**Comment by nasserman** [comment]
how i can use this patch? i install sylius according [this](https://docs.sylius.com/en/latest/book/installation/installation.html) guide and get that error again.

**Comment by lchrusciel** [comment]
Hey Nasser,

It appeared that something pops up to the Kamil, as Sylius is not tagged yet: https://github.com/Sylius/Sylius/tags. The best option would be to wait today until we will tag a new version.

However, you may change this line: https://github.com/Sylius/Sylius-Standard/blob/1.9/composer.json#L23 on your local, new installation to "dev-master" to use this fix right away

**Comment by pamil** [comment]
v1.8.11 and v1.9.2 were just tagged

**Comment by nasserman** [comment]
THANKS A LOT.

**Comment by cavasinf** [comment]
Same case here.
Fresh install `v1.12.8` without sample data on MySQL.

As soon `APP_DEBUG` is off we cannot create any simple Product.
But in debug mode, it works as untended.

>[2023-07-25T11:32:56.222373+02:00] request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\NotNullConstraintViolationException: "An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null" at /var/www/html/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 114 {"exception":"[object] (Doctrine\\DBAL\\Exception\\NotNullConstraintViolationException(code: 1048): An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null at /var/www/html/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:114)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1048): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null at /var/www/html/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'position' cannot be null at /var/www/html/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} []

![image](https://github.com/Sylius/Sylius/assets/25293190/7649fde5-3d5c-4b18-99e6-80b6e6daae8b)

![image](https://github.com/Sylius/Sylius/assets/25293190/bf531072-ac19-4a2c-9126-70eac0eba8e0)

**Comment by laurentdabbb** [comment]
Hi @cavasinf ,

Sylius team is aware of this issue, and has added a conflict last week for "stof/doctrine-extensions-bundle": "1.8.0" relative to your issue : https://github.com/Sylius/Sylius/pull/15154

Upgrade to Sylius 1.12.9 and your issue will be resolved ;-)

**Comment by cavasinf** [comment]
Wow you are fast @laurentdabbb 🚀 .
Upgrading to `1.12.9` is indeed the solution.

Thank you!


<details>
  <summary>Conflict while upgrading</summary>  
But a simple update is not that "simple".

`composer require sylius/sylius:^1.12.9`

```
  Problem 1
    - sylius/sylius 1.13.x-dev requires guzzlehttp/psr7 ^2.5 -> found guzzlehttp/psr7[2.5.0, 2.5.x-dev, 2.6.x-dev] but the package is fixed to 1.9.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - sylius/sylius[v1.12.9, ..., 1.12.x-dev] conflict with doctrine/orm 2.15.2 || 2.15.3.
    - lchrusciel/api-test-case v5.2.0 requires doctrine/orm ^2.5 -> satisfiable by doctrine/orm[2.15.3].
    - lchrusciel/api-test-case is locked to version v5.2.0 and an update of this package was not requested.
    - Root composer.json requires sylius/sylius ^1.12.9 -> satisfiable by sylius/sylius[v1.12.9, 1.12.x-dev, 1.13.x-dev]
```

I've done what was necessary on our end.
</details>
